### ISSUE TITLE
Split dataset

### ISSUE THREAD

**Issue by CaiSudao** [KevinMusgrave/powerful-benchmarker#40]

**Comment by CaiSudao** [comment]
I want to test all data at the same time, include training data, validation data and testing data. I don't know how to achieve this. Could you help me?

**Comment by KevinMusgrave** [comment]
This should work:
```
python run.py <your other flags> --eval_reference_set compared_to_sets_combined --splits_to_eval [train, val, test]
```
This will compute the accuracy for the following:
| query | reference |
| --- | --- |
| train | train+val+test |
| val | train+val+test |
| test | train+val+test |

The only limitation right now is that it must compute the accuracy for the train and val sets (the first 2 rows), even if you only want the test set accuracy (the 3rd row).

**Comment by CaiSudao** [comment]
Thanks a lot. You are so kind. I will try the 3rd row condition first.

**Comment by CaiSudao** [comment]
I used other datasets to do the experiment. The P@1 result about 100%. The training dataset includes 23 classes and each class 100 images. The testing dataset is the same of the training dataset. If the dataset is too easy?

**Comment by CaiSudao** [comment]
MAP@R and RP are about 50%

**Comment by KevinMusgrave** [comment]
Is the accuracy for ```compared_to_sets_combined``` different from the default setting (which is ```compared_to_self```)?

Did you use the default setting for ```split_manager```?

**Comment by CaiSudao** [comment]
I just used the compared_to_self setting. Maybe I could try the compared_to_sets_combined to compare. Yes, I used the default setting for split_manager.

**Comment by KevinMusgrave** [comment]
Yeah if you do ```compared_to_sets_combined```, the accuracy should drop at least a little bit. If you've already trained your model, you can use the ```--evaluate``` flag. The results for ```compared_to_self``` and ```compared_to_sets_combined``` will be kept separate.
```
python run.py --experiment_name <blah> --evaluate --eval_reference_set compared_to_sets_combined --splits_to_eval [train, val, test]
```

**Comment by CaiSudao** [comment]
OK. Thank you very much.
